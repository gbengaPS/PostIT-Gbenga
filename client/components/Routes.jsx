import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import Signup from './authentication/Signup';
import Login from './authentication/Login';
import Dashboard from './dashboard/Dashboard';
import NotFound from './errors/NotFound';
import Group from './group/Group';
import Welcome from './Home';
import Recover from './password/RecoverPassword';
import ChangePassword from './password/ChangePassword';
import MailSent from './password/MailSent';
import AuthHOC from './authentication/AuthHOC';
import NewGroup from './group/CreateGroup';
import Addmembers from './group/AddMembers';
import ErrorPage from './errors/Error';

/**
 * @description Routes component
 *
 * @returns {jsx} -jsx representation of the component
 */
const Routes = () => (
  <BrowserRouter>
    <Switch>
      <Route exact path="/" component={Welcome} />
      <Route path="/signup" component={Signup} />
      <Route path="/login" component={Login} />
      <Route path="/password/reset" component={Recover} />
      <Route path="/password/update" component={ChangePassword} />
      <Route path="/email/sent" component={MailSent} />
      <Route path="/dashboard" component={AuthHOC(Dashboard)} />
      <Route exact path="/group/:groupId/addmembers"
        component={AuthHOC(Addmembers)} />
      <Route exact path="/group/create" component={AuthHOC(NewGroup)} />
      <Route exact path="/group/:groupId" component={AuthHOC(Group)} />
      <Route exact path="/error" component={AuthHOC(ErrorPage)} />
      <Route path="*" component={NotFound} />
    </Switch>
  </BrowserRouter>
);

export default Routes;
