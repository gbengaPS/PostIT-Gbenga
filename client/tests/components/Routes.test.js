import React from 'react';
import 'mock-local-storage';
import Routes from '../../components/Routes';

describe('Routes component', () => {
  const wrapper = shallow(<Routes />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
