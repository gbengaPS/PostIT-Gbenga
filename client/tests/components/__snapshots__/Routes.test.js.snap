// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/signup"
        />
        <Route
          component={[Function]}
          path="/login"
        />
        <Route
          component={[Function]}
          path="/password/reset"
        />
        <Route
          component={[Function]}
          path="/password/update"
        />
        <Route
          component={[Function]}
          path="/email/sent"
        />
        <Route
          component={[Function]}
          path="/dashboard"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/group/:groupId/addmembers"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/group/create"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/group/:groupId"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/error"
        />
        <Route
          component={[Function]}
          path="*"
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/signup"
          />,
          <Route
            component={[Function]}
            path="/login"
          />,
          <Route
            component={[Function]}
            path="/password/reset"
          />,
          <Route
            component={[Function]}
            path="/password/update"
          />,
          <Route
            component={[Function]}
            path="/email/sent"
          />,
          <Route
            component={[Function]}
            path="/dashboard"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/group/:groupId/addmembers"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/group/create"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/group/:groupId"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/error"
          />,
          <Route
            component={[Function]}
            path="*"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/password/reset",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/password/update",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/email/sent",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/dashboard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/group/:groupId/addmembers",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/group/create",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/group/:groupId",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/signup"
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/password/reset"
          />
          <Route
            component={[Function]}
            path="/password/update"
          />
          <Route
            component={[Function]}
            path="/email/sent"
          />
          <Route
            component={[Function]}
            path="/dashboard"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/group/:groupId/addmembers"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/group/create"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/group/:groupId"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/error"
          />
          <Route
            component={[Function]}
            path="*"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/signup"
            />,
            <Route
              component={[Function]}
              path="/login"
            />,
            <Route
              component={[Function]}
              path="/password/reset"
            />,
            <Route
              component={[Function]}
              path="/password/update"
            />,
            <Route
              component={[Function]}
              path="/email/sent"
            />,
            <Route
              component={[Function]}
              path="/dashboard"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/group/:groupId/addmembers"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/group/create"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/group/:groupId"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/error"
            />,
            <Route
              component={[Function]}
              path="*"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/password/reset",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/password/update",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/email/sent",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/group/:groupId/addmembers",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/group/create",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/group/:groupId",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
